quinn-proto/src/connection.rs

```diff
@@ -71,6 +73,12 @@ pub struct Connection {
     events: VecDeque<Event>,
     /// Number of local connection IDs that have been issued in NEW_CONNECTION_ID frames.
     cids_issued: u64,
+    /// Packet number spaces: initial, handshake, 1-RTT
```

Comment:
I'm a little disappointed that my `VecDeque` structure didn't survive this! Is random access to spaces really so important? My intuition was that most access will want highest space anyway, and I liked not having to deal with the `Option`s.

Reply:
Your intuition is right in the sense of number of times accessed over the course of a typical connection, but it looks like few accesses are statically guaranteed to always want the highest space per se. The congestion control logic and `next_packet` may both need to use an earlier space if packet loss occurs during the transition, and all that leaves are the unconditional 1-RTT accesses (which look approximately the same regardless), `make_probe`, and `make_close`.

Reply:
I can probably abstract out the unwrapping for accesses via `self.space[_mut](...)`, though, looks like those are always expected to succeed.